CMakeLists.txt

```diff
@@ -154,17 +170,19 @@ if(Boost_FOUND)
             INTERFACE_LINK_LIBRARIES ${Boost_LIBRARIES})
     endif()
 
-    # link the whole user code library to Boost and JSON
-    target_link_libraries(lib PUBLIC Boost::filesystem Boost::program_options)
-
-    # then link it all to the executables
-    target_link_libraries(formatter PUBLIC Boost::filesystem lib)
-    target_link_libraries(tcp_example PUBLIC Boost::filesystem lib)
-    target_link_libraries(forwarding_example PUBLIC Boost::filesystem lib)
-    target_link_libraries(cli_example PUBLIC Boost::filesystem lib)
-    target_link_libraries(commanding_example PUBLIC Boost::filesystem lib)
-    target_link_libraries(test_spw_crc PUBLIC Boost::filesystem lib)
-    target_link_libraries(test PUBLIC Boost::filesystem lib)
+    # link the user libraries to Boost
+    target_link_libraries(libcom PUBLIC Boost::filesystem Boost::program_options)
+    target_link_libraries(libsubsys PUBLIC Boost::filesystem Boost::program_options)
+    target_link_libraries(libutil PUBLIC Boost::filesystem Boost::program_options)
+   
+    # then link them all to the executables
+    target_link_libraries(formatter PUBLIC Boost::filesystem libcom libsubsys libutil)
+    target_link_libraries(tcp_example PUBLIC Boost::filesystem libcom libsubsys libutil)
+    target_link_libraries(forwarding_example PUBLIC Boost::filesystem libcom libsubsys libutil)
```

Comment:
Change to `target_link_libraries(forwarding_example PUBLIC Boost::filesystem libsubsys libcom libutil)` for some reason